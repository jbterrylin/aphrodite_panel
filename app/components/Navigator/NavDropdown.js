/**
*
* NavDropdown
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Dropdown, { DropdownTrigger, DropdownContent } from 'react-simple-dropdown';
import './simple-dropdown.scss';

const DropdownContainer = styled(DropdownContent)`
    position: absolute;
    background: ${(props) => props.theme.secondary_bg};
    width: 500px;
    height: 300px;
    right: 16px;
`;

function NavDropdown(props) {
    return (
        <Dropdown className="NavDropdown-component nav-link text-capitalize account-dropdown" ref="dropdown">
            <DropdownTrigger>
                <span className={'account-dropdown__name ' + (props.item.iconClass ? props.item.iconClass : 'account-dropdown__name ')} title={props.item.text}></span>
            </DropdownTrigger>
            <DropdownContainer>
                <div className="account-dropdown__identity account-dropdown__segment">
                    Signed in as <strong>username here</strong>
                </div>
                <ul className="account-dropdown__quick-links account-dropdown__segment text-left">
                    <li className="account-dropdown__link">
                        <a className="account-dropdown__link__anchor" role="button" onClick={props.handleLinkClick}>
                            Your profile
                        </a>
                    </li>
                    <li className="account-dropdown__link">
                        <a className="account-dropdown__link__anchor" role="button" onClick={props.handleLinkClick}>
                            Your stars
                        </a>
                    </li>
                    <li className="account-dropdown__link">
                        <a className="account-dropdown__link__anchor" role="button" onClick={props.handleLinkClick}>
                            Explore
                        </a>
                    </li>
                    <li className="account-dropdown__link">
                        <a className="account-dropdown__link__anchor" role="button" onClick={props.handleLinkClick}>
                            Help
                        </a>
                    </li>
                </ul>
                <ul className="account-dropdown__management-links account-dropdown__segment">
                    <li className="account-dropdown__link">
                        <a className="account-dropdown__link__anchor" role="button" onClick={props.handleLinkClick}>
                            Settings
                        </a>
                    </li>
                    <li className="account-dropdown__link">
                        <a className="account-dropdown__link__anchor" onClick={props.handleLinkClick}>
                            Sign out
                        </a>
                    </li>
                </ul>
            </DropdownContainer>
        </Dropdown>
    );
}

NavDropdown.propTypes = {
    handleLinkClick: PropTypes.func,
    item: PropTypes.object,
};

export default NavDropdown;
