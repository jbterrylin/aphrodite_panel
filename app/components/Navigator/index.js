/**
*
* Navigator
*
*/

import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

function Navigator(props) {
    const menu = props.items.map((item) => (
        <NavLink className="nav-item nav-link" key={item.name} to={item.link}>{item.name}</NavLink>
    ));

    return (
        <div>
            <nav className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded">
                <button className="navbar-toggler navbar-toggler-right" type="button">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        {menu}
                    </div>
                </div>
            </nav>
        </div>
    );
}

Navigator.propTypes = {
    items: PropTypes.array.isRequired,
};

export default Navigator;
