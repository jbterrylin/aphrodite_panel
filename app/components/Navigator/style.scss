.gami-navbar {
    border: 1px solid transparent;

    .gami-navbar-nav {
        margin: 0;

        .nav-item {
            float: left;
        }
    }
}