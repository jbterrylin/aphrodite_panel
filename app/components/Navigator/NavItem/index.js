/**
*
* NavItem
*
*/

import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import styled from 'styled-components';
// import styled from 'styled-components';

import NavDropdown from './NavDropdown';

const StyleNavi = styled.span`
    padding: 1rem;
    cursor: pointer;
    color: white;

    &:hover {
        color: ${(props) => props.theme.tertiary_color};
    }
`;

function NavItem(props) {
    if (props.data.type === 'internal_url') {
        return (
            <NavLink to={props.data.url} className="nav-link text-capitalize" title={props.data.title}>
                <StyleNavi className={props.data.iconClass ? props.data.iconClass : ''}>
                    {props.data.text}
                </StyleNavi>
            </NavLink>
        );
    } else if (props.data.type === 'external_url') {
        return (
            <a href={props.data.url} className="nav-link text-capitalize" title={props.data.title}>
                <StyleNavi className={props.data.iconClass ? props.data.iconClass : ''}>{props.data.text}</StyleNavi>
            </a>
        );
    } else if (props.data.type === 'dropdown') {
        return (
            <NavDropdown item={props.data}>
                <StyleNavi className={(props.data.iconClass ? props.data.iconClass : 'dropdown__name ')} title={props.data.title}></StyleNavi>
            </NavDropdown>
        );
    } else if (props.data.type === 'exec_function') {
        return (
            <a
                onClick={props.data.handleLinkClick}
                role="button"
                tabIndex="0"
                className="nav-link text-capitalize"
                title={props.data.title}
            >
                <StyleNavi className={props.data.iconClass ? props.data.iconClass : ''}>{props.data.text}</StyleNavi>
            </a>
        );
    } else if (props.data.type === 'render') {
        return props.data.render(props);
    }

    return (
        <span className="nav-link text-capitalize" title={props.data.title}>
            <StyleNavi className={props.data.iconClass ? props.data.iconClass : ''}>{props.data.text}</StyleNavi>
        </span>
    );
}

NavItem.propTypes = {
    data: PropTypes.object.isRequired,
};

export default NavItem;
