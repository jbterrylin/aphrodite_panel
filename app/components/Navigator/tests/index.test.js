import React from 'react';
import { shallow } from 'enzyme';

import Navigator from '../index';

describe('<Navigator />', () => {
    const navItems = [
        { link: '/mall', name: 'Malls' },
        { link: '/flagship', name: 'Flagship Stores' },
    ];
    const wrapper = shallow(<Navigator items={navItems} />);

    it('should render Navigator Correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a nav element', () => {
        expect(wrapper.find('nav').length).toEqual(1);
    });
});
