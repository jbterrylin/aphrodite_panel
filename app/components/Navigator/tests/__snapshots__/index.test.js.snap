// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigator /> should render Navigator Correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <nav
        className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded"
    >
        <button
            className="navbar-toggler navbar-toggler-right"
            type="button"
        >
            <span
                className="navbar-toggler-icon"
            />
        </button>
    </nav>
</div>,
  "nodes": Array [
    <div>
      <nav
            className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded"
      >
            <button
                  className="navbar-toggler navbar-toggler-right"
                  type="button"
            >
                  <span
                        className="navbar-toggler-icon"
                  />
            </button>
      </nav>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Navigator
        items={
                Array [
                        Object {
                          "link": "/mall",
                          "name": "Malls",
                        },
                        Object {
                          "link": "/flagship",
                          "name": "Flagship Stores",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "items": Array [
            Object {
              "link": "/mall",
              "name": "Malls",
            },
            Object {
              "link": "/flagship",
              "name": "Flagship Stores",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <nav
                    className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded"
          >
                    <button
                              className="navbar-toggler navbar-toggler-right"
                              type="button"
                    >
                              <span
                                        className="navbar-toggler-icon"
                              />
                    </button>
          </nav>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <nav
                    className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded"
          >
                    <button
                              className="navbar-toggler navbar-toggler-right"
                              type="button"
                    >
                              <span
                                        className="navbar-toggler-icon"
                              />
                    </button>
          </nav>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Navigator
    items={
        Array [
            Object {
              "link": "/mall",
              "name": "Malls",
            },
            Object {
              "link": "/flagship",
              "name": "Flagship Stores",
            },
          ]
    }
/>,
}
`;
